lighthouse-cli/run.js

```diff
@@ -26,15 +26,20 @@ const _RUNTIME_ERROR_CODE = 1;
 const _PROTOCOL_TIMEOUT_EXIT_CODE = 67;
 
 /**
- * exported for testing
+ * Parse Chrome Flags string using yargs into individual arguments, so they can be passed to
+ * chrome-launcher.
+ * The flags are scanned for any disabled default flags (specified as `${flag}=false`). If any
+ * default flags are disabled, we ignore all default flags then add back any default flags that
+ * were not disabled.
+ * This function is exported for testing.
  * @param {string} flags
- * @return {Array<string>}
+ * @return {{chromeFlags: Array<string>, ignoreDefaultFlags: boolean}}
  */
 function parseChromeFlags(flags = '') {
   const parsed = yargsParser(
```

Comment:
This line is what's responsible for the failed test (well, in addition to the Chrome Launcher bug).

The value of `flags` passed in is:
```
--disable-translate=false --disable-extensions=false --disable-background-networking=false --disable-sync=false --metrics-recording-only=false --disable-default-apps=false --mute-audio=false --no-first-run=false --remote-debugging-port=40051=false --disable-setuid-sandbox=false --user-data-dir=/tmp/lighthouse.2KVKTKX=false about:blank=false --remote-debugging-port=40051=false --disable-setuid-sandbox=false --user-data-dir=/tmp/lighthouse.2KVKTKX=false about:blank=false
```

and the value of `parsed is:
```
{
    _:[
        'about:blank=false',
        'about:blank=false'
    ],
    'disable-translate':'false',
    'disable-extensions':'false',
    'disable-background-networking':'false',
    'disable-sync':'false',
    'metrics-recording-only':'false',
    'disable-default-apps':'false',
    'mute-audio':'false',
    'no-first-run':'false',
    'remote-debugging-port':[
        '40051=false',
        '40051=false'
    ],
    'disable-setuid-sandbox':[
        'false',
        'false'
    ],
    'user-data-dir':[
        '/tmp/lighthouse.2KVKTKX=false',
        '/tmp/lighthouse.2KVKTKX=false'
    ]
}
```

As you can see, appending `=false` doesn't work so well when the flags have values.

Reply:
ah yes, we are indeed blocked on that bug then :) we'll try to merge it ASAP!